seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/sink/ElasticsearchSinkWriter.java

```diff
@@ -99,29 +105,27 @@ public Optional<ElasticsearchCommitInfo> prepareCommit() {
     public void abortPrepare() {
     }
 
-    public void bulkEsWithRetry(EsRestClient esRestClient, List<String> requestEsList, int maxRetry) {
-        for (int tryCnt = 1; tryCnt <= maxRetry; tryCnt++) {
-            if (requestEsList.size() > 0) {
-                String requestBody = String.join("\n", requestEsList) + "\n";
-                try {
+    public void bulkEsWithRetry(EsRestClient esRestClient, List<String> requestEsList) {
+        try {
+            RetryUtils.retryWithException(() -> {
+                if (requestEsList.size() > 0) {
+                    String requestBody = String.join("\n", requestEsList) + "\n";
                     BulkResponse bulkResponse = esRestClient.bulk(requestBody);
-                    if (!bulkResponse.isErrors()) {
-                        break;
+                    if (bulkResponse.isErrors()) {
+                        throw new BulkElasticsearchException("bulk es error: " + bulkResponse.getResponse());
                     }
-                } catch (Exception ex) {
-                    if (tryCnt == maxRetry) {
-                        throw new BulkElasticsearchException("bulk es error,try count=%d", ex);
-                    }
-                    log.warn(String.format("bulk es error,try count=%d", tryCnt), ex);
+                    return bulkResponse;
                 }
-
-            }
+                return null;
+            }, retryMaterial);
+        } catch (Exception e) {
+            throw new RuntimeException("ElasticSearch execute batch statement error", e);
```

Comment:
Please use `SeaTunnelException` and we have an issue to discuss the unified Exception define. If you interested in it, you can see https://github.com/apache/incubator-seatunnel/pull/3045.

Reply:
done.